VideoLocker/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -135,79 +136,63 @@ public void onClick(View v) {
         tryToSetUIInteraction(true);
     }
 
-    public void showAgreement(RegistrationAgreement agreement) {
-        boolean isInAppEULALink = false;
-        try {
-            Uri uri = Uri.parse(agreement.getLink());
-            if (uri.getScheme().equals("edxapp")
-                    && uri.getHost().equals("show_eula")) {
-                isInAppEULALink = true;
-            }
-        } catch (Exception ex) {
-            logger.error(ex);
-        }
-
-        if (isInAppEULALink) {
-            // show EULA license that is shipped with app
-            environment.getRouter().showWebViewDialog(this, getString(R.string.eula_file_link), getString(R.string.end_user_title));
-        } else {
-            // for any other link, open agreement link in a webview container
-            environment.getRouter().showWebViewDialog(this, agreement.getLink(), agreement.getText());
-        }
+    public void showAgreement() {
+        environment.getRouter().showWebViewDialog(this, getString(R.string.eula_file_link), getString(R.string.end_user_title));
     }
 
     private void setupRegistrationForm() {
-        try {
-            RegistrationDescription form = environment.getServiceManager().getRegistrationDescription();
-
-            LayoutInflater inflater = getLayoutInflater();
+        new GetRegistrationDescriptionTask(this) {
 
-            List<RegistrationFormField> agreements = new ArrayList<>();
+            @Override
+            protected void onSuccess(RegistrationDescription registrationDescription) throws Exception {
+                LayoutInflater inflater = getLayoutInflater();
+
+                List<RegistrationFormField> agreements = new ArrayList<>();
+
+                for (RegistrationFormField field : registrationDescription.getFields()) {
+                    if (field.getFieldType().equals(RegistrationFieldType.CHECKBOX)) {
+                        // this is agreement field
+                        // this must be added at the end of the form
```

Comment:
Hmm, I see, it's because we group the optional fields at the end, and so we are also forced to move the agreement to the end, else it would appear before the optional fields.

I wonder if instead we should generalize this to "any field that appears after ALL the optional fields, will appear below the optional fields section.

Or maybe it's not worth worrying about, not sure.